spec/opentok/broadcasts_spec.rb

```diff
@@ -223,4 +223,40 @@
     expect(response).not_to be_nil
   end
 
+  xcontext "for many broadcasts" do
+    it "should return all broadcasts", :vcr => { :erb => { :version => OpenTok::VERSION + "-Ruby-Version-#{RUBY_VERSION}-p#{RUBY_PATCHLEVEL}"} } do
+      broadcast_list = broadcasts.all
+      expect(broadcast_list).to be_an_instance_of OpenTok::BroadcastList
+      expect(broadcast_list.total).to eq 6
+      expect(broadcast_list.count).to eq 6
+    end
+
+    it "should return broadcasts with an offset", :vcr => { :erb => { :version => OpenTok::VERSION + "-Ruby-Version-#{RUBY_VERSION}-p#{RUBY_PATCHLEVEL}"} } do
+      broadcast_list = broadcasts.all :offset => 3
+      expect(broadcast_list).to be_an_instance_of OpenTok::BroadcastList
+      expect(broadcast_list.total).to eq 3
+      expect(broadcast_list.count).to eq 3
+    end
+
+    it "should return count number of broadcasts", :vcr => { :erb => { :version => OpenTok::VERSION + "-Ruby-Version-#{RUBY_VERSION}-p#{RUBY_PATCHLEVEL}"} } do
+      broadcast_list = broadcasts.all :count => 2
+      expect(broadcast_list).to be_an_instance_of OpenTok::BroadcastList
+      expect(broadcast_list.count).to eq 2
+      expect(broadcast_list.count).to eq 2
+    end
+
+    it "should return part of the broadcasts when using offset and count", :vcr => { :erb => { :version => OpenTok::VERSION + "-Ruby-Version-#{RUBY_VERSION}-p#{RUBY_PATCHLEVEL}"} } do
+      broadcast_list = broadcasts.all :count => 4, :offset => 2
+      expect(broadcast_list).to be_an_instance_of OpenTok::BroadcastList
+      expect(broadcast_list.count).to eq 4
+      expect(broadcast_list.count).to eq 4
+    end
+
+    it "should return session broadcasts", :vcr => { :erb => { :version => OpenTok::VERSION + "-Ruby-Version-#{RUBY_VERSION}-p#{RUBY_PATCHLEVEL}"} } do
+      broadcast_list = broadcasts.all :sessionId => session_id
+      expect(broadcast_list).to be_an_instance_of OpenTok::BroadcastList
+      expect(broadcast_list.total).to eq 3
+      expect(broadcast_list.count).to eq 3
+    end
+  end
```

Comment:
For all of these tests, `all` should be called on `broadcast` (singular) not `broadcasts` (plural). See [line 19](https://github.com/opentok/OpenTok-Ruby-SDK/blob/142ef849cba11715a31b3513e8d87eec376de793/spec/opentok/broadcasts_spec.rb#L19) where the variable is defined:

```ruby
let(:broadcast) { opentok.broadcasts }
```